operator/v1alpha1/types.go

```diff
@@ -150,3 +150,31 @@ type DelegatedAuthorization struct {
 	// disabled indicates that authorization should be disabled.  By default it will use delegated authorization.
 	Disabled bool `json:"disabled,omitempty"`
 }
+
+// StaticPodOperatorStatus is status for controllers that manage static pods.  There are different needs because individual
+// node status must be tracked.
+type StaticPodOperatorStatus struct {
+	OperatorStatus `json:",inline"`
+
+	// latestDeploymentID is the deploymentID of the most recent deployment
+	LatestDeploymentID int32 `json:"latestDeploymentID"`
+
+	// nodeStatuses track the deployment values and errors across individual nodes
+	NodeStatuses []NodeStatus `json:"nodeStatuses"`
+}
+
+// NodeStatus provides information about the current state of a particular node managed by this operator.
+type NodeStatus struct {
+	// nodeName is the name of the node
+	NodeName string `json:"nodeName"`
+
+	// currentDeploymentID is the ID of the most recently successful deployment
```

Comment:
You should probably call this Generation (if it's a monotonic increasing number) or Version (if you want more semantics about ordering).  ID is unusual

Reply:
> You should probably call this Generation (if it's a monotonic increasing number) or Version (if you want more semantics about ordering). ID is unusual

ok.  It is not related to the object generation though.